preprocessing.py

```diff
@@ -75,35 +75,43 @@ def preprocess_csmith_file(
             "-E",
         ] + additional_flags
         lines = utils.run_cmd(cmd).split("\n")
-        marker_range = find_marker_decl_range(lines, marker_prefix)
-        platform_main_end_line = find_platform_main_end(lines)
-        if not platform_main_end_line:
-            raise PreprocessError("Couldn't find 'platform_main_end'")
-        marker_decls = lines[marker_range[0] : marker_range[1]]
-
-        lines = lines[platform_main_end_line + 1 :]
-        lines = remove_print_hash_value(remove_platform_main_begin(lines))
-        lines = (
-            marker_decls
-            + [
-                "typedef unsigned int size_t;",
-                "typedef signed char int8_t;",
-                "typedef short int int16_t;",
-                "typedef int int32_t;",
-                "typedef long long int int64_t;",
-                "typedef unsigned char uint8_t;",
-                "typedef unsigned short int uint16_t;",
-                "typedef unsigned int uint32_t;",
-                "typedef unsigned long long int uint64_t;",
-                "int printf (const char *, ...);",
-                "void __assert_fail (const char *__assertion, const char *__file, unsigned int __line, const char *__function);",
-                "static void",
-                "platform_main_end(uint32_t crc, int flag)",
-            ]
-            + list(lines)
-        )
 
-        return "\n".join(lines)
+        start_patterns = [
+            re.compile("^extern.*"),
+            re.compile("^typedef.*"),
+            re.compile("^struct.*"),
+        ]
+        taint_patterns = [
+            re.compile(".*__access__.*"),  # LLVM doesn't know about this
+            re.compile(".*__malloc__.*"),
+            re.compile(
+                ".*_[F|f]loat[0-9]{1,3}x{0,1}.*"
+            ),  # https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html#Floating-Types
+            re.compile(".*__asm__.*"),  # CompCert has problems
+        ]
+        final_code: list[str] = []
+        linepos_in_code = 0
+        run = 0
+        tainted = False
+        for line in lines:
+            for p in start_patterns:
+                if p.match(line):
+                    if not tainted:
+                        final_code.extend(
+                            lines[linepos_in_code : linepos_in_code + run]
+                        )
+                    linepos_in_code += run
+                    run = 0
+                    tainted = False
+            for p in taint_patterns:
```

Comment:
Why do we only check for for `tainted_patterns` after copying and not before?

Is not possible to do something like this:
```
for i, line in enumerate(lines):
  if i in lines_to_skip:
    continue
  for p in taint_patterns:
    if p.match(line):
      new_lines_to_skip = #figure out where the beginning and end of the decl are
      lines_to_skip |= new_lines_to_skip
  return "\n".join(line for i, line in lines_to_skip if i not in lines_to_skip"
```
This code would be a bit easier to understand (and change in the future)